File: fun.cs
using System;

class CSharpCalculator
{
    static void Main(string[] args)
    {
        try
        {
            Console.Write("Enter the first number: ");
            double num1 = Convert.ToDouble(Console.ReadLine());

            Console.Write("Enter an operator (+, -, *, /): ");
            char op = Convert.ToChar(Console.ReadLine());

            Console.Write("Enter the second number: ");
            double num2 = Convert.ToDouble(Console.ReadLine());

            double result = 0;

            switch (op)
            {
                case '+':
                    result = num1 + num2;
                    break;
                case '-':
                    result = num1 - num2;
                    break;
                case '*':
                    result = num1 * num2;
                    break;
                case '/':
                    if (num2 == 0)
                    {
                        Console.WriteLine("Error: Cannot divide by zero.");
                        return;
                    }
                    result = num1 / num2;
                    break;
                default:
                    Console.WriteLine("Error: Invalid operator.");
                    return;
            }
            Console.WriteLine($"Result: {result}");
        }
        catch (FormatException)
        {
            Console.WriteLine("Error: Invalid number format.");
        }
    }
}

File: fun.hs
import Text.Read (readMaybe)

main :: IO ()
main = do
    putStrLn "Enter the first number:"
    num1Str <- getLine
    let maybeNum1 = readMaybe num1Str :: Maybe Double

    putStrLn "Enter an operator (+, -, *, /):"
    opStr <- getLine

    putStrLn "Enter the second number:"
    num2Str <- getLine
    let maybeNum2 = readMaybe num2Str :: Maybe Double

    case (maybeNum1, opStr, maybeNum2) of
        (Just n1, "+", Just n2) -> print (n1 + n2)
        (Just n1, "-", Just n2) -> print (n1 - n2)
        (Just n1, "*", Just n2) -> print (n1 * n2)
        (Just n1, "/", Just n2) ->
            if n2 == 0
            then putStrLn "Error: Cannot divide by zero."
            else print (n1 / n2)
        _ -> putStrLn "Invalid input or operator."
